### ISSUE TITLE
Make the boot screen nicer

### ISSUE THREAD

**Issue by ListenerJubatus** [stepmania/stepmania#535]
On Windows especially, the splash screen looks like a relic of Windows 95 with its old fonts and plain progress bar. In the words of best pony, it needs to be about 20% cooler. Personally, I think a more "vendor neutral" look for it could promote consistency between platforms.

![new sm splash](https://cloud.githubusercontent.com/assets/6346433/6935593/36369bf2-d805-11e4-8d34-0b8a4f40fc60.png)

Little something like this perhaps?

**Comment by Wallacoloo** [comment]
I see that you edited your post to make your reference more subtle :P

I agree that your mockup looks nicer than the default theme. I've seen different themes (e.g. Simply Love) modify the splash logo, so perhaps there's already a way for them to customize the loading bar and text without digging into the C++?

Not to derail the thread, but if we're considering improving the splash screen I think it's worth at least *considering* removing it all together. To my knowledge, the only thing being done during the splash screen is indexing the song and course metadata. It seems to me that this could just be done in the background after presenting the full GUI, and the user could instead be presented with a splash only when they attempt to enter the song selection screen before all songs have been indexed.

I haven't tested the recent improvements to the loading system, but previously, my 1300 songs took about 10 seconds to be indexed, and it usually takes me about 7-8 seconds after being presented with a GUI before I've gone through enough menus (or rather, fade-through-black transitions) to be presented with the song selection screen. So loading the metadata in the background could reduce the time it takes to begin a match from 18 seconds to just 10 seconds, in my case.

**Comment by kyzentun** [comment]
The only things that can be customized about the loading window right now are the splash logo ("Graphics/Common splash.png") and the icon ("Graphics/Common window icon.png").  It's going to stay that way because it's not worth the time to do anything besides maybe change the font and colors for reasons explained in detail below.

Removing the loading window altogether:
This cannot be done without seriously restructuring Stepmania.  Attempting to do so would take a huge amount of work and create a section of Stepmania theming that has various restrictions that would have to be forced by the engine, and thus would limit theming to exactly what exists now.

On my personal machine, after the loading window finishes, 9 seconds are spent just loading the font textures used in the overlay screens.  (yes, the font system should be rewritten, that's a separate issue).  If the loading window is removed, and loading takes place after the main window comes up, that's 9 seconds where the player stares at a black screen (or worse, no window at all) because nothing else can be shown yet.  When someone starts an app, and *nothing* happens for 9 seconds, they assume it's broken, because that is often the case.  So they flail around doing random things, then have to spend more time cleaning up whatever they did during the flailing.  This is not acceptable.
So delay loading the overlay screens until after a loading screen.  Well, the loading screen can't use any font until it gets a message from the background thread that the font textures are done loading.  All it can do is show a little loading image.  Then it has to somehow deal with handling the message for font loading completing, and switching to song loading progress.
Song loading must happen before the theme is allowed to do anything because the initial screen is controlled by two things:  
1.  The InitialScreen metric in the Common section.
  It's perfectly possible to make a theme that kicks the player straight into gameplay, no fooling around with picking songs or options or anything.  All those layers of screen transitions wasting your time?  Completely optional for the theme, and only there because someone did it that way 10 years ago and doing it in a way that doesn't waste time hasn't caught on.
2.  The TestInitialScreen preference in Preferences.ini.
  This is used by themers to skip straight to some random screen they want to test the look of, so they don't have to go through the trouble of playing through the game.  The screens in the engine all have code in them to fill whatever data they need with test data specifically for this preference to work at all and not crash instantly.
Thus, song loading is forced to happen before the theme does anything.
If song loading happens in the main window, on a special screen that the engine forces the theme to start on before forcing a transition to the actual initial screen set by the metric, then here's how it looks:
The themer has to create a screen that has no text until the fonts are done loading.  Then it has to show the progress.  Since themers can't deal with the no text problem, that means the engine has to enforce it somehow, by hard coding the actors in C++, and not loading any lua.  Which means the theme can't control what actors are on the screen at all, except to supply a loading image.  That's what we have now.  All this work to restructure how loading is done, and it comes full circle exactly back to where we are now.


References:
My machine is a 3.2 GHz AMD Phenom from 2011.  1.5 TB SATA drive from the same year, don't remember the manufacturer, but it was under $100 at the time.
We don't have useful stats on the performance of common player machines, so my load time could be longer or shorter than average.

drunken-shame is a joke theme I made that demonstrates (almost) skipping straight to gameplay after loading finishes.  I put in a little info screen to explain controls just for kindness.  https://github.com/kyzentun/drunken-shame

The overlay screens serve various useful purposes, like supplying the debug menu that is vital to dealing with many things.  That's why they exist at all times.

Any programmer wishing to profile startup time can use the macros defined in RageTimer.h, or use some professionally made profiling software.  Just make sure you clear your filesystem cache before each test, or your data will be confounded by the OS caching stuff for faster filesystem access.

**Comment by SolraBizna** [comment]
Aleph One's pre-level "blank screen" loading times got incredibly long as the normal size of texture packs increased. A few years back they added a customizable loading screen, using a separate (and much less flexible, therefore more lightweight) system from the rest of the game's interface customizations. A themer provides an image to display, a position and style in which to draw the loading screen text, and a rectangle and colors with which to the loading bar. (It is a little more complicated than that, but that's the gist.)

Perhaps something like this would fit in StepMania?
